pkg/api/store/nodetemplate/store.go

```diff
@@ -25,6 +26,24 @@ type Store struct {
 	CloudCredentialLister corev1.SecretLister
 }
 
+func Wrap(store types.Store, npLister v3.NodePoolLister, secretLister corev1.SecretLister) types.Store {
+	return &transform.Store{
+		Store: &Store{
+			Store:                 store,
+			NodePoolLister:        npLister,
+			CloudCredentialLister: secretLister,
+		},
+		Transformer: filterLegacyTemplates,
+	}
+}
+
+func filterLegacyTemplates(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}, opt *types.QueryOptions) (map[string]interface{}, error) {
+	if data["namespaceId"] != namespace.NodeTemplateGlobalNamespace {
```

Comment:
It could be useful to enqueue the nodetemplate to get it migrated if it's seen in here